lib/https.js

```diff
@@ -218,7 +218,8 @@ exports.request = function request(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
-  } else if (options instanceof url.URL) {
+  } else if (options && options[searchParamsSymbol] &&
+             options[searchParamsSymbol][searchParamsSymbol]) {
```

Comment:
Would a `require('internal/url').isURL()` utility function be worthwhile to make this more convenient? Or would that kill any perf gain? I imagine wanting to use similar checks elsewhere.